plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisherITest.java

```diff
@@ -362,6 +363,26 @@ void shouldHonorWithChecksContextRecordIssues() {
                 output -> assertThat(output.getTitle()).isPresent().get().isEqualTo("No new issues, 6 total."));
     }
 
+    /**
+     * Verifies that {@link WarningChecksPublisher} uses issue category when type is not provided.
+     */
+    @Test
+    void shouldFallbackToIssueCategory() {
+        FreeStyleProject project = createFreeStyleProject();
+        enableWarnings(project, createTool(new MsBuild(), "msbuild.log"));
+
+        buildSuccessfully(project);
+
+        copySingleFileToWorkspace(project, "msbuild.log");
+        Run<?, ?> run = buildSuccessfully(project);
+
+        WarningChecksPublisher publisher = new WarningChecksPublisher(getResultAction(run), TaskListener.NULL, null);
+        ChecksDetails details = publisher.extractChecksDetails(AnnotationScope.PUBLISH_NEW_ISSUES);
+
+        assertThat(details.getOutput().get().getChecksAnnotations().get(0))
+                .hasFieldOrPropertyWithValue("title", Optional.of("C4101"));
```

Comment:
`C4101` seems to be a strange category, shouldn't this be the type actually? Maybe that is the root cause...

Reply:
Yes as I written in the description most of the MsBuild warnings have no type and only a category. The MsBuild parser sets the category instead of the type, when no category is provided. But changing the MsBuild parser would be a breaking change, or would you prefer to fix the issue inside the MsBuld parser?

Reply:
I think you change here makes sense anyway as some other parsers might have an empty type as well. 

But in general it would also make sense to reconsider the existing implementation of the MsBuild parser and see if the mapping can be improved. I'm using it not on my own so I am not sure if such a change would be helpful.

Reply:
Issue category and type are both shown inside the "issue view". The other use case I am aware of is ignoring / filtering issues, where its only a mater of configuration. But at least its a little confusing.